service/history/ndc/workflow_resetter.go

```diff
@@ -795,8 +798,22 @@ func reapplyEvents(
 			}
 			reappliedEvents = append(reappliedEvents, event)
 		default:
```

Comment:
cc @yux0 Is the `isDuplicate(event)` check still useful today? I know it's in-memory and doesn't provide any kind of dedup guarantee. Shall we deprecated that or there's a plan for making it real?

Reply:
yes. this is best effort. We should dedup them based on the version histories